AADRoleEligibilityScheduleRequest - : The Role assignment already exists - on Administrative Unit as DirectoryScopeId

### Description of the issue

Hi guys,
I run a devops pipeline with this module configured like that:

- Name: "<group-to-assingn-NameDescription>"
    Action: "AdminAssign"
    DirectoryScopeId: "/administrativeUnits/<AUId>"  
    DependsOn: "[AADGroup]<group-to-assingn>"
    Ensure: "Present"
    IsValidationOnly: False
    Principal: "<group-to-assingn-Name>"
    PrincipalType: "Group"
    RoleDefinition: "Privileged Authentication Administrator"
    ScheduleInfo:
      startDateTime: "2024-07-05T11:08:33Z"
      expiration:
        type: "noExpiration"
  
  So the first run of the pipeline is ok and the role correctly assigned.    
 From a "second run" of the pipeline we get this error:
 
 ##[error][RoleAssignmentExists] : The Role assignment already exists.
    + CategoryInfo          : InvalidOperation: ({ Headers = , b...heduleRequest }:) [], CimException
    + FullyQualifiedErrorId : RoleAssignmentExists,Microsoft.Graph.Beta.PowerShell.Cmdlets.NewMgBetaRoleManagementDire 
   ctoryRoleEligibilityScheduleRequest_CreateExpanded
    + PSComputerName        : localhost
VERBOSE: [fv-az524-586]: LCM:  [ End    Set      ]  
[[AADRoleEligibilityScheduleRequest]<group-to-assingn-NameDescription>::[EntraID]EntraID_Configuration]  in 
5.5520 seconds.
##[error]The PowerShell DSC resource 


No problem with the DirectoryScopeId on "/"
I think it is a bug.

Thank you in advance.

### Microsoft 365 DSC Version

1.24.904.1

### Which workloads are affected

Azure Active Directory (Entra ID)

### The DSC configuration

```powershell
AADRoleEligibilityScheduleRequest:
 - Name: "<group-to-assingn-NameDescription>"
    Action: "AdminAssign"
    DirectoryScopeId: "/administrativeUnits/<AUId>"  
    DependsOn: "[AADGroup]<group-to-assingn>"
    Ensure: "Present"
    IsValidationOnly: False
    Principal: "<group-to-assingn-Name>"
    PrincipalType: "Group"
    RoleDefinition: "Privileged Authentication Administrator"
    ScheduleInfo:
      startDateTime: "2024-07-05T11:08:33Z"
      expiration:
        type: "noExpiration"
```


### Verbose logs showing the problem

```powershell
##[error][RoleAssignmentExists] : The Role assignment already exists.
    + CategoryInfo          : InvalidOperation: ({ Headers = , b...heduleRequest }:) [], CimException
    + FullyQualifiedErrorId : RoleAssignmentExists,Microsoft.Graph.Beta.PowerShell.Cmdlets.NewMgBetaRoleManagementDire 
   ctoryRoleEligibilityScheduleRequest_CreateExpanded
    + PSComputerName        : localhost
VERBOSE: [fv-az524-586]: LCM:  [ End    Set      ]  
[[AADRoleEligibilityScheduleRequest]<group-to-assingn-NameDescription>::[EntraID]EntraID_Configuration]  in 
5.5520 seconds.
##[error]The PowerShell DSC resource
```


### Environment Information + PowerShell Version

```powershell
git version 2.45.2.windows.1
Task PowerShell Version: 2.245.1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,9 @@
   * Fixes a bug where 3P apps could not be assigned by DisplayName for both
     IncludeApplications and ExcludeApplications
     FIXES [#5390](https://github.com/microsoft/Microsoft365DSC/issues/5390)
+* AADRoleEligibilityScheduleRequest
+  * FIXES [#3787](https://github.com/microsoft/Microsoft365DSC/issues/3787)
+  * FIXES [#5089](https://github.com/microsoft/Microsoft365DSC/issues/5089)
 * EXOATPBuiltInProtectionRule, EXOEOPProtectionRule
   * Fixed issue where empty arrays were being compared incorrectly to null
     strings

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_AADRoleEligibilityScheduleRequest/MSFT_AADRoleEligibilityScheduleRequest.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_AADRoleEligibilityScheduleRequest/MSFT_AADRoleEligibilityScheduleRequest.psm1
@@ -190,7 +190,7 @@
                         $schedule = $instance
                     }
                 }
-                [Array]$request = Get-MgBetaRoleManagementDirectoryRoleEligibilityScheduleRequest -Filter "PrincipalId eq '$PrincipalId' and RoleDefinitionId eq '$($schedule.RoleDefinitionId)'" | Sort-Object -Property CompletedDateTime -Descending
+                [Array]$request = Get-MgBetaRoleManagementDirectoryRoleEligibilityScheduleRequest -Filter "PrincipalId eq '$PrincipalId'" | Where-Object -FilterScript {$_.RoleDefinitionId -eq $schedule.RoleDefinitionId} | Sort-Object -Property CompletedDateTime -Descending
 `
                 if ($request.Length -gt 1)
                 {

--- a/Tests/Unit/Microsoft365DSC/Microsoft365DSC.AADRoleEligibilityScheduleRequest.Tests.ps1
+++ b/Tests/Unit/Microsoft365DSC/Microsoft365DSC.AADRoleEligibilityScheduleRequest.Tests.ps1
@@ -55,6 +55,7 @@ Describe -Name $Global:DscHelper.DescribeHeader -Fixture {
             Mock -CommandName Get-MgBetaRoleManagementDirectoryRoleEligibilitySchedule -MockWith {
                 return @{
                     Id          = '12345-12345-12345-12345-12345'
+                    RoleDefinitionId = "12345"
                 }
             }
 

